examples/other/background_subtraction.py

```diff
@@ -0,0 +1,189 @@
+radius = np.sqrt(X**2 + Y**2)
+im *= (1 - radius[..., np.newaxis] / radius.max())
+im = im.astype(np.uint8)
+
+canvas_size = im.shape[0], im.shape[1]
+canvas = WgpuCanvas(size=canvas_size, max_fps=999)
+renderer = gfx.renderers.WgpuRenderer(canvas, show_fps=True)
+scene = gfx.Scene()
+camera = gfx.OrthographicCamera(canvas_size[0], canvas_size[1])
+camera.local.y = canvas_size[1] / 2
+camera.local.scale_y = -1
+camera.local.x = canvas_size[0] / 2
+controller = gfx.PanZoomController(camera, register_events=renderer)
+
+
+image_texture = gfx.Texture(im, 
+                            dim=2, 
+                            generate_mipmaps=True
+                            )
+
+class BackGroundRemovedImageMaterial(gfx.ImageBasicMaterial):
+    """
+    An image that has the background removed.
+    """
+    uniform_type = dict(
+        gfx.ImageBasicMaterial.uniform_type,
+        background_level="u4",
+    )
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.background_level = 0
+
+    @property
+    def background_level(self):
+        return self.uniform_buffer.data["background_level"]
+
+    @background_level.setter
+    def background_level(self, value):
+        self.uniform_buffer.data["background_level"] = int(value)
+        self.uniform_buffer.update_range()
+
+class BackGroundRemovedImage(gfx.Image):
+    pass
+
+@register_wgpu_render_function(BackGroundRemovedImage, BackGroundRemovedImageMaterial)
+class BackGroundRemovedImageShader(ImageShader):
+    def __init__(self, wobject):
+        super().__init__(wobject)
+        material = wobject.material
+
+    def get_code(self):
+        code = super().get_code()
+        code = code.replace(
+            """
+    let value = sample_im(varyings.texcoord.xy, sizef);
+""",
+            """
+    let new_size = sizef.xy / pow(2., f32(u_material.background_level));
+    let texcoords_u = vec2<f32>(varyings.texcoord.xy * new_size);
+    let texcoords_img = vec2<f32>(varyings.texcoord.xy * sizef.xy);
+    var f: vec2<f32> = fract(min(texcoords_u + 0.5, vec2<f32>(new_size)));
+    var f_img: vec2<f32> = fract(min(texcoords_img + 0.5, vec2<f32>(sizef.xy)));
+
+    // Get the background value via bilinear interpolation
+    var b0: vec4<f32> = mix(
+        textureLoad(t_img, 
+            vec2<i32>(texcoords_u + vec2<f32>(-0.5, -0.5)), 
+            i32(u_material.background_level)),
+        textureLoad(t_img, 
+            vec2<i32>(texcoords_u + vec2<f32> (0.5, -0.5)), 
+            i32(u_material.background_level)),
+        f.x,
+    );
+
+    var b1: vec4<f32> = mix(
+        textureLoad(t_img, 
+            vec2<i32>(texcoords_u + vec2<f32>(-0.5, 0.5)), 
+            i32(u_material.background_level)),
+        textureLoad(t_img, 
+            vec2<i32>(texcoords_u + vec2<f32>( 0.5, 0.5)), 
+            i32(u_material.background_level)),
+        f.x,
+    );
+    let background = mix(b0, b1, f.y);
```

Comment:
Instead of using `textureLoad` to read 4 pixels and combine them, you can also use `textureSampleLevel` to let the hardware do the interpolation. Unless I'm overlooking a reason for using `textureLoad` here.